lib/ansible/module_utils/common/validation.py

```diff
@@ -381,7 +381,7 @@ def check_type_str(value, allow_conversion=True, param=None, prefix=''):
     if isinstance(value, string_types):
         return value
 
-    if allow_conversion:
+    if allow_conversion and value is not None:
```

Comment:
Previous to the change in lib/ansible/module_utils/common/parameters.py, `None` string options were never converted to `'None'` because the type enforcement was skipped for all None values. However with my change, the type is now validated for required options and options that have a non-None default.

Returning 'None' would be a breaking change for `apt_repository`, which relies on None as a valid value for the **required** str option [`repo`](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/apt_repository_module.html#parameter-repo).

I waffled between two ways to fix this. Either do this, and update the tests since `None` will now fail type conversion (what I did), or return None as-is above (requires no update to the tests).